### DISCUSSION TITLE
`write_scrollback_buffer:open` does not work with terminal applications on Linux

### DISCUSSION THREAD

**Discussion by carrascomj** [ghostty-org/ghostty#4054] [Q&A] [answered]
I am making a new discussion to highlight this, but this has been previously reported on comments to issues: https://github.com/ghostty-org/ghostty/issues/3334#issuecomment-2564296601 and https://github.com/ghostty-org/ghostty/issues/2504#issuecomment-2564947390. I have tested this on Linux.

`write_scrollback_buffer:paste` **works** as expected. The temporary file has `text/plain` as the reported filetype by xdg-mime and it can be openen by xdg-open even when set to a terminal application.

`write_scrollback_buffer:open` **does not work** as expected. When `text/plain` is to be opened by a terminal application and the action is triggered, nothing occurs. No temporary file is written to `/tmp` either. This works when a non-terminal application is used instead; for example, `xdg-mime default firefox.desktop text/plain` will open the scrollback file in firefox.

As reported before, opening ghostty from a different terminal emulator and then triggering `write_scrollback_buffer:open` opens the history in the parent terminal emulator when a terminal app is set.

I can also verify that the action `open_config` has the same behavior; it does not work with terminal applications on Linux (https://github.com/ghostty-org/ghostty/issues/3334),

#### `ghostty --version`

```
Ghostty 1.0.0-dev+0000000

Version
  - version: 1.0.0-dev+0000000
  - channel: tip
Build Config
  - Zig version: 0.13.0
  - build mode : builtin.OptimizeMode.ReleaseFast
  - app runtime: apprt.Runtime.gtk
  - font engine: font.main.Backend.fontconfig_freetype
  - renderer   : renderer.OpenGL
  - libxev     : main.Backend.io_uring
  - GTK version:
    build      : 4.16.7
    runtime    : 4.16.7
  - libadwaita : enabled
    build      : 1.6.2
    runtime    : 1.6.2
```

**Comment by 00-kat** [comment, marked as answer]
Probably a duplicate of #3585.

**Reply by 00-kat** [reply]
Consider sending some of the information here in #3585, and moving discussion there too, to avoid having multiple discussions of the same topic.
